neqo-transport/Cargo.toml

```diff
@@ -20,6 +20,7 @@ neqo-common = { path = "../neqo-common" }
 neqo-crypto = { path = "../neqo-crypto" }
 qlog = { workspace = true }
 smallvec = { version = "1.11", default-features = false }
+static_assertions = { version = "1.1", default-features = false }
```

Comment:
This is only used in `neqo-transport/src/connection/tests/datagram.rs`, right? If so, how about moving it to `[dev-dependencies]`?